worker/instancepoller/worker.go

```diff
@@ -346,7 +346,7 @@ func (u *updaterWorker) pollGroupMembers(groupType pollGroupType) error {
 	}
 
 	infoList, err := u.config.Environ.Instances(u.callContext, instList)
-	if err != nil && err != environs.ErrPartialInstances {
+	if err != nil && !(err == environs.ErrPartialInstances || err == environs.ErrNoInstances) {
```

Comment:
When do we get NoInstances? Are we passing in a empty list? or just ending up with nothing that matches?

Reply:
The ec2 [provider](https://github.com/juju/juju/blob/develop/provider/ec2/environ.go#L924-L931) in particular checks whether all of the results in the list are nil and returns ErrNoInstances if all are nil or ErrPartialInstances if at least one entry is nil.

Reply:
So this might happen if say, we are polling the long poll group and **all** instances have been deleted but we haven't yet processed the machine deletion events to remove them from the poll group... I guess it's quite rare for this to happen but in any case, the instance poller shouldn't need to crash if it does occur.